repos/system_upgrade/el7toel8/actors/vimmigrate/actor.py

```diff
@@ -0,0 +1,59 @@
+from leapp.actors import Actor
+from leapp.libraries.common.reporting import report_generic
+from leapp.models import Report, VimModel
+from leapp.tags import ApplicationsPhaseTag, IPUWorkflowTag
+
+from os import access, W_OK
+
+class VimMigrate(Actor):
+    """
+    Actor for adding two directives into vim configuration files to simulate 
+    the same behavior as vim-7.4 has.
+    """
+
+    name = 'vim_migrate'
+    consumes = (VimModel,)
+    produces = (Report,)
+    tags = (ApplicationsPhaseTag, IPUWorkflowTag)
+
+    """ Macros for adding into configuration file"""
+    _changes = [
+                'let skip_defaults_vim=1',
+                'set t_BE='
+              ]
+
+
+    def insert_string(self, string, path):
+        """
+        Insert string into a file
+
+        :param str string: string which will be inserted
+        :param str path: string representing path
+        """
+
+        if not access(path, W_OK):
+            raise OSError('File {} does not have writing permissions.'.format(path))
+
+        try:
+            with open(path, 'a') as f:
+                f.write(string + '\n')
+        except IOError:
+            raise IOError('Error during writing to file {}.'.format(path))
+
+    def process(self):
+        for configs_list in self.consume(VimModel):
+            if not configs_list:
```

Comment:
Did you actually ever encounter this? Because if this is happening the framework has a bug
The vim model has vim_configs which is the only thing you should check, but then again if this is empty why to send the message at all?
Think about it

Reply:
It's removed already. Just waiting for new push-force. Marking this as resolved as `VimModel` has been removed completely.